sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataReaderDAO.java

```diff
@@ -627,10 +644,21 @@ protected MultiTableScanResult computeNext() {
                 while (_iter.hasNext()) {
                     // Get the next rows from the grouping iterator.  All rows in the returned Iterable
                     // are from the same Cassandra wide row (in other words, they share the same key).
-                    Iterable<Row> rows = _iter.next();
+                    final Iterable<Row> rows = _iter.next();
+
+                    // filter the rows if a cutOff time is specified.
+                    Iterable<Row> filteredRows = rows;
+                    if (timeUUID != null) {
+                        filteredRows = () -> Iterators.filter(rows.iterator(), new Predicate<Row>() {
+                            @Override
+                            public boolean apply(@Nullable Row row) {
+                                return (TimeUUIDs.compare(getChangeId(row), timeUUID) < 0);
+                            }
+                        });
+                    }
```

Comment:
Minor:  Can use Java 8 lambda for predicate:

`filteredRows = () -> Iterators.filter(rows.iterator(), row -> (TimeUUIDs.compare(getChangeId(row), timeUUID) < 0));`